File: FureyLib/Misc or Not Mine/Go/Boilerplates/struct.go
// .go
// Name
// Creator

package MyStruct

import "math"

// STRUCTURE VARIABLES

/** Template structure script. */
type MyStruct struct {
	/** Example x variable. */
	X float32

	/** Example y variable. */
	Y float32

	// New variables here
}

// STRUCTURE CONSTRUCTORS

/** Default constructor. */
func NewStruct(x float32, y float32) MyStruct {
	return MyStruct{x, y}
}

// New constructors here

// STRUCTURE METHODS

/** Returns the magnitude of this structure. */
func (self *MyStruct) Magnitude() float32 {
	if self.X == 0.0 && self.Y == 0.0 {
		return 0.0
	}
	return float32(math.Sqrt(float64(self.X*self.X + self.Y*self.Y)))
}

/** Normalizes this structure. */
func (self *MyStruct) Normalize() *MyStruct {
	var magnitude float32 = self.Magnitude()
	if magnitude == 0.0 {
		return self
	}
	self.X /= magnitude
	self.Y /= magnitude
	return self
}

// New methods here

File: FureyLib/Misc or Not Mine/Zig/Boilerplates/struct.zig
// .zig
// Name
// Creator

const std = @import("std");

/// Template structure script.
pub const Struct = struct {
    // VARIABLES

    /// Example x variable.
    x: f32,

    /// Example y variable.
    y: f32,

    // New variables here


    // CONSTRUCTORS AND DESTRUCTOR

    /// Initializes a new structure.
    pub fn init(x: f32, y: f32) Struct {
        const self = Struct{
            .x = x,
            .y = y
        };
        return self;
    }

    // New constructors here

    /// Properly destroys the given structure.
    pub fn deinit(self: *Struct) void {
        self.x = 0;
        self.y = 0;
    }


    // METHODS

    /// Returns the magnitude of the given structure.
    pub fn magnitude(self: *const Struct) f32 {
        if (self.x == 0 and self.y == 0) {
            return 0;
        }
        return std.math.sqrt(self.x * self.x + self.y * self.y);
    }

    /// Normalizes the given structure.
    pub fn normalize(self: *Struct) *Struct {
        const mag = self.magnitude();
        if (mag == 0) {
            return self;
        }
        self.x /= mag;
        self.y /= mag;
        return self;
    }

    // New methods here
};


